src/Oscar.jl

```diff
@@ -33,57 +33,80 @@ const jll_deps = String["Antic_jll", "Arb_jll", "Calcium_jll", "FLINT_jll", "GAP
                         "libpolymake_julia_jll", "libsingular_julia_jll", "msolve_jll",
                         "polymake_jll", "Singular_jll"];
 
-function _lookup_git_branch(dir::AbstractString)
-   if length(Sys.which("git")) != nothing &&
-         isdir(joinpath(dir,".git"))
+function _lookup_commit_from_cache(url::AbstractString, tree::AbstractString)
+   if Sys.which("git") != nothing
       try
-         ref = cd(dir) do
-            readchomp(`git rev-parse --abbrev-ref HEAD`)
+         path = Pkg.Types.add_repo_cache_path(url)
+         if isdir(path)
+            commit = readchomp(`sh -c "git -C $path log --oneline --all --pretty='tree %T;%H' | grep \"^tree $tree\" | cut -d\; -f2 | head -n1"`)
```

Comment:
So what's wrong with something like this:
```suggestion
            commit = readchomp(`git -C $path rev-parse HEAD`)
```
(Of course if that work, you might as well remove this line and change the next one to use `HEAD` instead of `$commit`)

However, given that the code used `git rev-parse --abbrev-ref HEAD` before, I assume there is a problem with using `HEAD`, but what is it? Is `HEAD` simply missing? I only have two dirs in `~/.julia/clones` but they both have a `HEAD`

Reply:
`HEAD` in such a bare repo will just point to the latest commit of the default branch, but we might be tracking a different branch instead of `master`.

For example I now did `add Polymake#bl/jl18`, then `]st` shows:
```
  [d720cf60] Polymake v0.7.1 `https://github.com/oscar-system/Polymake.jl.git#bl/jl18`
```
but the clone still has:
```
9242f08 (HEAD -> master, origin/master, origin/HEAD, cache/heads/master) Merge pull request #398 from oscar-system/bl/banner
```

But versioninfo shows:
```
    Polymake.jl          v0.7.1 - #bl/jl18, c69199a -- 2022-03-23 15:41:05 +0100
```

Unfortunately the `PackageInfo` struct only contains the tree-hash (and the ref-name) but no info about the commit which is why I added that code, and it will probably be very rarely executed as I don't think many people use `]add` with a branch.

Reply:
I wasn't concerned about performance, just confused what's going on. Now I understand, thank you for the explanation. Perhaps a sentence could be added as a comment? Like:
> We can't use HEAD here as it just points to the latest commit on the default
> branch, which may differ from what is checked out. All we have is the the git
> tree which is checked out, which we map back to a commit hash here.